libraries/shared/storage/diff_repository_test.go

```diff
@@ -274,35 +286,56 @@ var _ = Describe("Storage diffs repository", func() {
 
-	Describe("MarkChecked", func() {
-		It("marks a diff as checked", func() {
-			fakeRawDiff := types.RawDiff{
-				HashedAddress: test_data.FakeHash(),
-				BlockHash:     test_data.FakeHash(),
-				BlockHeight:   rand.Int(),
-				StorageKey:    test_data.FakeHash(),
-				StorageValue:  test_data.FakeHash(),
-			}
-			fakePersistedDiff := types.PersistedDiff{
-				RawDiff:   fakeRawDiff,
+	Describe("Changing the diff status", func() {
+		var fakePersistedDiff types.PersistedDiff
+		BeforeEach(func() {
+			fakePersistedDiff = types.PersistedDiff{
+				RawDiff:   fakeStorageDiff,
 				ID:        rand.Int63(),
-				Checked:   false,
+				Status:    "new",
 				EthNodeID: db.NodeID,
 			}
-			_, insertErr := db.Exec(`INSERT INTO public.storage_diff (id, block_height, block_hash,
-				hashed_address, storage_key, storage_value, checked, eth_node_id) VALUES ($1, $2, $3, $4, $5, $6, $7, $8)`,
-				fakePersistedDiff.ID, fakeRawDiff.BlockHeight, fakeRawDiff.BlockHash.Bytes(),
-				fakeRawDiff.HashedAddress.Bytes(), fakeRawDiff.StorageKey.Bytes(), fakeRawDiff.StorageValue.Bytes(),
-				fakePersistedDiff.Checked, fakePersistedDiff.EthNodeID)
-			Expect(insertErr).NotTo(HaveOccurred())
+			insertTestDiff(fakePersistedDiff, db)
+		})
 
-			err := repo.MarkChecked(fakePersistedDiff.ID)
+		It("marks a diff as transformed", func() {
+			err := repo.MarkTransformed(fakePersistedDiff.ID)
 
 			Expect(err).NotTo(HaveOccurred())
-			var checked bool
-			checkedErr := db.Get(&checked, `SELECT checked FROM public.storage_diff WHERE id = $1`, fakePersistedDiff.ID)
-			Expect(checkedErr).NotTo(HaveOccurred())
-			Expect(checked).To(BeTrue())
+			var status string
+			getStatusErr := db.Get(&status, `SELECT status FROM public.storage_diff WHERE id = $1`, fakePersistedDiff.ID)
+			Expect(getStatusErr).NotTo(HaveOccurred())
+			Expect(status).To(Equal("transformed"))
+		})
+
+		It("marks a diff as unrecognized", func() {
+			err := repo.MarkUnrecognized(fakePersistedDiff.ID)
+
+			Expect(err).NotTo(HaveOccurred())
+			var status string
+			getStatusErr := db.Get(&status, `SELECT status FROM public.storage_diff WHERE id = $1`, fakePersistedDiff.ID)
+			Expect(getStatusErr).NotTo(HaveOccurred())
+			Expect(status).To(Equal("unrecognized"))
+		})
+
+		It("marks a diff as noncanonical", func() {
+			err := repo.MarkNoncanonical(fakePersistedDiff.ID)
+
+			Expect(err).NotTo(HaveOccurred())
+			var status string
+			getStatusErr := db.Get(&status, `SELECT status FROM public.storage_diff WHERE id = $1`, fakePersistedDiff.ID)
+			Expect(getStatusErr).NotTo(HaveOccurred())
+			Expect(status).To(Equal("noncanonical"))
+		})
+
+		It("marks a diff as unwatched", func() {
+			err := repo.MarkUnwatched(fakePersistedDiff.ID)
+
+			Expect(err).NotTo(HaveOccurred())
+			var status string
+			getStatusErr := db.Get(&status, `SELECT status FROM public.storage_diff WHERE id = $1`, fakePersistedDiff.ID)
+			Expect(getStatusErr).NotTo(HaveOccurred())
+			Expect(status).To(Equal("unwatched"))
```

Comment:
nbd but I wonder if we should wrap these strings (e.g. `"new"`, `"unrecognized"`, `"noncanonical"`, `"unwatched"`, `"transformed"`) in vars so that we can change them in one place if we change the enum in the DB

Reply:
i like this idea!